src/telemetry.rs

```diff
@@ -35,10 +48,77 @@ pub fn setup_logging() {
 #[cfg(not(feature = "console"))]
 pub fn setup_logging() {
     Lazy::force(&APPLICATION_START_TIME);
-    let filter_layer = tracing_subscriber::EnvFilter::try_from_default_env()
+    let filter = tracing_subscriber::EnvFilter::try_from_default_env()
         .or_else(|_| tracing_subscriber::EnvFilter::try_new("info"))
         .unwrap();
-    tracing_subscriber::registry()
-        .with(tracing_subscriber::fmt::layer().with_filter(filter_layer))
-        .init();
+    let (filter_layer, reload_handle) =
+        reload::Layer::new(tracing_subscriber::fmt::layer().with_filter(filter));
+    match LOG_HANDLE.set(LogHandle {
+        handle: reload_handle,
+    }) {
+        Ok(_) => {}
+        Err(_) => {
+            eprintln! {"setup log handler failed\n"};
+        }
+    };
+    tracing_subscriber::registry().with(filter_layer).init();
+}
+
+// a handle to get and set the log level
+type BoxLayer = tracing_subscriber::fmt::Layer<tracing_subscriber::Registry>;
+pub(crate) type FilteredLayer = filter::Filtered<BoxLayer, EnvFilter, Registry>;
+pub struct LogHandle {
+    handle: reload::Handle<FilteredLayer, Registry>,
+}
+
+pub fn set_level(level_str: String) -> Result<String, Error> {
```

Comment:
```suggestion
pub fn set_level(level_str: String) -> Result<LevelFilter, Error> {
```

would be more idiomatic. We shouldn't be encoding random strings, but use typed values. The caller can translate this to a message